### PR TITLE
Item with doubled discount category shouldn't be clone

### PR DESCRIPTION
**Pull request by maciekjablonski**
Thanks for @czemiello for report issue in #655.

### CODE REVIEW

**Comment by patrys** [comment]
The change looks ok, but I'm not sure what that test is checking. It's obvious that it would fail but it's not obvious what that would mean for the store.

File: tests/test_discounts.py [hunk 1]
```diff
@@ -246,3 +249,24 @@ def test_sale_applies_to_correct_products(product_class):
         net=5, currency='USD')
     with pytest.raises(NotApplicable):
         sale.modifier_for_product(sec_variant.product)
+
+
+@pytest.mark.django_db
+def test_get_category_variants_and_prices_product_with_many_categories(cart, default_category, product_in_stock):
+    """Test error: get_category_variants_and_prices return duplicated items. It cause double(or more) percentage discount"""
```
Comment (patrys) [inline tests/test_discounts.py]: This will become meaningless as soon as we merge this branch. Please drop the docstring.

**Comment by maciekjablonski** [comment]
@patrys  i fixed styles

File: tests/test_discounts.py [hunk 2]
```diff
@@ -252,10 +252,12 @@ def test_sale_applies_to_correct_products(product_class):
 
 
 @pytest.mark.django_db
-def test_get_category_variants_and_prices_product_with_many_categories(cart, default_category, product_in_stock):
-    """Test error: get_category_variants_and_prices return duplicated items. It cause double(or more) percentage discount"""
+def test_get_category_variants_and_prices_product_with_many_categories(cart, default_category,
+                                                                       product_in_stock):
+    # Test error: get_category_variants_and_prices return duplicated items.
```
Comment (krzysztofwolski) [inline tests/test_discounts.py]: We don't need to test errors. Describe what this function tests and what is expected output.

**Pull request merged**